### ISSUE TITLE
[Spike] Time lock for tokens

### ISSUE THREAD

**Issue by credbull** [credbull/credbull-defi#2]
Spike Success Criteria
- Lock tokens for X period of time (days/months/etc).  
- Connect the time lock to a Safe Wallet or ERC-4626 Vault.

**Comment by pasviegas** [comment]
For the Safe implementation I think we don't need a spike as it is fairly well tested (with readable tests)
- https://github.com/gnosis/token-lock/blob/master/packages/contracts/test/TokenLock.spec.ts

**Comment by lucasia** [comment]
> For the Safe implementation I think we don't need a spike as it is fairly well tested (with readable tests)
> 
> * https://github.com/gnosis/token-lock/blob/master/packages/contracts/test/TokenLock.spec.ts

sorry spike maybe wasn't clear.  i wanted to try connecting the Time lock to a Vault.  so, not just the Time lock in isolation.  

the spikes so far do result in duplicate testing that we can probably remove.  but it's also about the learning journey. :)

**Comment by pasviegas** [comment]
For Safe, we need to decide which version we will use as the apis for each differ. I chose to use 1.3.0 for this spike as the 1.4.1 is in "alpha".

To achieve the timelock I used an existing [modifier](https://github.com/gnosis/zodiac-modifier-delay) created by gnosis and attached it to a safe. It seats in the middle between the safe and a module so any transaction a module tries to execute is queued for when the initial cooldown time is complete and only then can be executed.
